opencog/atoms/proto/FactoryLoader.cc

```diff
@@ -0,0 +1,31 @@
+
+#include <opencog/atoms/proto/ProtoAtom.h>
+#include <opencog/atoms/proto/FloatValue.h>
+#include <opencog/atoms/proto/LinkValue.h>
+#include <opencog/atoms/proto/RandomStream.h>
+#include <opencog/atoms/proto/StringValue.h>
+#include <opencog/atoms/proto/ValueFactory.h>
+
+using namespace opencog;
+
+static __attribute__ ((constructor)) void init(void)
+{
+    valuefactory().register_factory(LINK_VALUE, (CreateProto) & (createLinkValue<std::vector<ProtoAtomPtr>>),
+                                    std::vector<std::type_index> {std::type_index(typeid(std::vector<ProtoAtomPtr>))});
+
```

Comment:
The above 4 lines should be copied into `LinkValue.cc`  .. similarly, the lines below should be copied to the corresponding `*.cc` files.